### PR TITLE
[build] Only upload artifacts for clang builds

### PR DESCRIPTION
**Pull request by ehsannas**

### CODE REVIEW

**Comment by ehsannas** [comment]
the Travis gcc builds failed because they didn't deploy.

**Comment by ehsannas** [comment]
https://travis-ci.org/github/microsoft/DirectXShaderCompiler/builds/679173633

**Comment by ehsannas** [comment]
this should fix it.

**Comment by pow2clk** [comment]
this is it I think: https://travis-ci.org/github/microsoft/DirectXShaderCompiler/builds/679070362 Looks green. Curious what happened.

**Comment by ehsannas** [comment]
Travis skips the `deploy` stage for pull-requests. So pull-requests will continue to be green, but the merge commit may be red.

**Comment by ehsannas** [comment]
What's odd is that the clang builds for the merge commit passed the deploy stage [successfully](https://travis-ci.org/github/microsoft/DirectXShaderCompiler/jobs/679173636#L12494) (no warnings no errors), but I don't see anything in the cloud bucket :(

**Comment by ehsannas** [comment]
from the log:
```
0.04s$ if [[ "${CC}" == "clang" ]]; then export ARTIFACTS_DIR=build-${TRAVIS_OS_NAME}-${DXC_BUILD_TYPE}; cd .. && mkdir ${ARTIFACTS_DIR}; cd ${ARTIFACTS_DIR} && mkdir bin && mkdir lib && mkdir include && cd ..; cp build/lib/libdxcompiler.so.3.7 ${ARTIFACTS_DIR}/lib/; cp build/bin/dxc ${ARTIFACTS_DIR}/bin/; cp -r include/dxc ${ARTIFACTS_DIR}/include; echo "${TRAVIS_COMMIT}" > ${ARTIFACTS_DIR}/GIT-COMMIT.txt; cat ${ARTIFACTS_DIR}/GIT-COMMIT.txt; ls ${ARTIFACTS_DIR}/*; fi
dpl_0
1.43s$ rvm $(travis_internal_ruby) --fuzzy do ruby -S gem install dpl
8.45s
dpl.1
Installing deploy dependencies
Successfully installed gstore-0.2.1
Successfully installed mime-types-data-3.2019.1009
Successfully installed mime-types-3.3.1
Successfully installed dpl-gcs-1.10.15
Parsing documentation for gstore-0.2.1
Installing ri documentation for gstore-0.2.1
Parsing documentation for mime-types-data-3.2019.1009
Installing ri documentation for mime-types-data-3.2019.1009
Parsing documentation for mime-types-3.3.1
Installing ri documentation for mime-types-3.3.1
Parsing documentation for dpl-gcs-1.10.15
Installing ri documentation for dpl-gcs-1.10.15
Done installing documentation for gstore, mime-types-data, mime-types, dpl-gcs after 2 seconds
4 gems installed
Logging in with Access Key: ****************2MMA
/home/travis/.rvm/gems/ruby-2.4.1/gems/gstore-0.2.1/lib/gstore/request.rb:76: warning: constant OpenSSL::Digest::Digest is deprecated
dpl.2
Preparing deploy
dpl.3
Deploying application
Done. Your build exited with 0.
```

File: .travis.yml [hunk 1]
```diff
@@ -102,4 +102,5 @@ deploy:
   on:
     repo: microsoft/DirectXShaderCompiler
     branch: master
+    condition: $CC=clang
```
Comment (tex3d) [inline .travis.yml:105]: This didn't work.  I don't know anything about the syntax used here, but should it be `==`?

Comment (ehsannas) [reply .travis.yml:105]: The Travis [documentation](https://docs.travis-ci.com/user/deployment) says we should use `=`. 
It looks like Travis is **not** uploading the artifacts (for any configuration) to the cloud storage (the bucket is still empty). And the Travis logs show no errors or warning. So there's no way to figure out what's going wrong... I'll abandon this effort for now to stop the linux builds from failing.

**Pull request merged**